source/agora/consensus/protocol/Nominator.d

```diff
@@ -544,6 +553,21 @@ extern(D):
                     fail_reason, data);
                 return ValidationLevel.kInvalidValue;
             }
+
+            bool is_same = this.ledger.isSameMissingValidatorsAsLocal(data);
+            if (is_same)
+            {
+                if (auto fail_reason = this.ledger.validateSlashingData(data))
+                {
+                    log.error("validateValue(): Preimage Validation failed: {}. Data: {}",
+                        fail_reason, data);
+                    return ValidationLevel.kInvalidValue;
+                }
```

Comment:
Add `return ValidationLevel. kFullyValidatedValue` here and then you won't need the `temp_level` anymore.